chain/chain/src/blocks_delay_tracker.rs

```diff
@@ -488,79 +488,87 @@ impl Chain {
         log_summary_style: LogSummaryStyle,
     ) -> Result<String, Error> {
         let chain_info = self.get_chain_processing_info();
-        let blocks_info = chain_info.blocks_info;
-        let use_colour = matches!(log_summary_style, LogSummaryStyle::Colored);
-        let paint = |colour: ansi_term::Colour, text: Option<String>| match text {
-            None => ansi_term::Style::default().paint(""),
-            Some(text) if use_colour => colour.bold().paint(text),
-            Some(text) => ansi_term::Style::default().paint(text),
+        let blocks_info = BlocksInfo {
+            blocks_info: chain_info.blocks_info,
+            use_colour: matches!(log_summary_style, LogSummaryStyle::Colored),
         };
 
-        // Returns a status line for each block - also prints what is happening to its chunks.
-        let next_blocks_log = blocks_info
-            .into_iter()
-            .flat_map(|block_info| {
-                let mut all_chunks_received = true;
-                let chunk_status_str = block_info
-                    .chunks_info
-                    .iter()
-                    .map(|chunk_info| {
-                        if let Some(chunk_info) = chunk_info {
-                            all_chunks_received &=
-                                matches!(chunk_info.status, ChunkProcessingStatus::Completed);
-                            match chunk_info.status {
-                                ChunkProcessingStatus::Completed => "✔",
-                                ChunkProcessingStatus::Requested => "⬇",
-                                ChunkProcessingStatus::NeedToRequest => ".",
-                            }
-                        } else {
-                            "X"
-                        }
-                    })
-                    .collect::<Vec<&str>>()
-                    .join("");
+        Ok(format!(
```

Comment:
Hm, this method overall is only ever used once, as an argument to `debug!`. Ideally I think that log statement is changed to use KVs, as described here, https://github.com/near/nearcore/blob/master/docs/practices/style.md#tracing but even without looking to make a change as large as this we could look into returning a [`fmt::Arguments`](https://doc.rust-lang.org/stable/std/fmt/struct.Arguments.html) (possibly by changing this method to use a callback closure, as `Arguments` has a lifetime) to avoid one further extra allocation.

Reply:
Done.